unit/java_bytecode/java_bytecode_specialize_generics/java_bytecode_specialize_generics.cpp

```diff
@@ -0,0 +1,299 @@
+/*******************************************************************\
+
+ Module: Unit tests for specializing generic methods
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_bytecode/java_bytecode_language.h>
+#include <java_bytecode/generate_java_generic_method.h>
+#include <util/namespace.h>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+SCENARIO(
+  "java_bytecode_specialize_generics",
+  "[core][java_bytecode][java_bytecode_specialize_generics]")
+{
+  symbol_tablet new_symbol_table=load_java_class("generics", ""
+    "./java_bytecode/java_bytecode_specialize_generics");
+
+  generate_java_generic_methodt instantiate_generic_method{};
+
+
+  GIVEN("Some class files with Generics")
+  {
+    WHEN("Methods with generic signatures")
```

Comment:
Nit: I'd prefer if this was more specific `WHEN("Generic method returning the generic type")`